lib/ansible/modules/crypto/luks_device.py

```diff
@@ -193,10 +231,41 @@ class Handler(object):
     def __init__(self, module):
         self._module = module
         self._lsblk_bin = self._module.get_bin_path('lsblk', True)
+        self._blkid_bin = self._module.get_bin_path('blkid', True)
 
     def _run_command(self, command):
         return self._module.run_command(command)
 
+    def get_device_by_uuid(self, uuid):
+        ''' Returns the device that holds UUID passed by user
+        '''
+        uuid = self._module.params['uuid']
+        if uuid is None:
+            return None
+
+        query = '%s=%s' % ('UUID', uuid)
+        result = self._run_command([self._blkid_bin, '-l', '-t', query])
```

Comment:
Fun fact: when I run this as my user, it returns nothing. When I run it as root, it works. On the other hand, `blkid --uuid <UUID>` also works when I run it as my user.

So why not use
```suggestion
        result = self._run_command([self._blkid_bin, '--uuid', query])
```
instead? That seems to be simpler. Also, you don't need any postprocessing, since it returns only the device name and nothing else.

Reply:
Nice, much better sir, thanks ;-D